src/rest/RequestHandler.js

```diff
@@ -136,8 +217,12 @@ class RequestHandler {
       // Handle ratelimited requests
       if (res.status === 429) {
         // A ratelimit was hit - this should never happen
-        this.manager.client.emit('debug', `429 hit on route ${request.route}`);
-        await Util.delayFor(this.retryAfter);
+        this.manager.client.emit('debug', `429 hit on route ${request.route}${res.sublimit ? ' for sublimit' : ''}`);
+        // If caused by a sublimit, wait it out here so other requests on the route can be handled
+        if (res.sublimit) {
+          await Util.delayFor(res.sublimit);
+          delete res.sublimit;
```

Comment:
Please do not use `delete`, also refer to the other request (regarding mutating external objects).